clients/client/src/main/java/com/dremio/nessie/client/ClientContentsApi.java

```diff
@@ -34,56 +36,46 @@ public ClientContentsApi(WebTarget target) {
     this.target = target;
   }
 
-  @Override
-  public Contents getContents(@NotNull ContentsKey key, String ref) throws NessieNotFoundException {
-    return target.path("contents").path(key.toPathString()).path(ref)
-                 .request()
-                 .accept(MediaType.APPLICATION_JSON_TYPE)
-                 .get()
-                 .readEntity(Contents.class);
-  }
 
   @Override
-  public Contents getContents(ContentsKey key) throws NessieNotFoundException {
+  public Contents getContents(@NotNull ContentsKey key, String ref) throws NessieNotFoundException {
     return target.path("contents").path(key.toPathString())
+                 .queryParam("ref", ref)
```

Comment:
How does this show up in the construction? It seems like we should reject the ref parameter if it is provided but has no value. Thoughts?

Reply:
So the client does not add `ref` query parameter if `ref` (the variable) is null

Reply:
Are you saying that `foo/?ref=&baz=bar` would be rejected? and that if you passed `(ref, null)` that it would just eliminate the ref parameter entirely? Just confirming that you were stating, not asking.

Thanks

Reply:
I guess there's the server behavior, and the client behavior.
* For the client, `queryParam("ref", null)` would remove `ref` from the list of parameters to be sent as part of the URL. To produce the URL you suggested, one would need to use `queryParam("ref", "")`
* For the server, since the argument is annotated with `@NotNull`, absence of the `ref` query parameter would throw a 400 Bad Request without invoking the method. An empty `ref` will invoke the method, with the implementation likely to send also 400 Bad Request as an empty reference is not valid. I guess we could constrain a bit further and use `@NotEmpty` or better.